src/androidTest/java/androidx/core/app/job/JobSchedulerTest.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.core.app.job
+
+import android.app.job.JobParameters
+import android.app.job.JobService
+import android.content.ComponentName
+import android.support.test.InstrumentationRegistry
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.concurrent.withLock
+
+class JobSchedulerTest {
+    @Test fun testScheduleJob() {
+        scheduleJob(context, ComponentName(context, TestService::class.java), 0) {
```

Comment:
Having to use the context twice here is weird. This function call does not read very well. Perhaps we should address #526 first. But beyond that I'm not sure I'm convinced we need to handle scheduling and job creation in a single function.

Reply:
I've updated the API to be
```kotlin
scheduleJob<TestService>(context, jobId) {
  /* ... */
}
```
which I think reads much nicer. Also, #526 won't change the function call in this case. Do you think this version makes it worth it to handle the scheduling and creation in a single function?

Reply:
I've updated the API again as per @rom4ek's suggestion and also added an upper bound on the type passed to the `ComponentName`. Please check out the **Benefits** section in the PR description to see why I think this would be worth it. Looking forward to hearing your thoughts!